fix: invalid behavior on sid/alternative in 5.3.4/99.5.4.5.1

sinrazon opened the pull request:
On Debian Sid / alternative distributions (such as Ubuntu), these checks would fail with `sid: integer expression expected`, which is to be expected since `$DEB_MAJ_VER` is set to a string, not an integer.

This small PR fixes that in a similar fashion to other checks in the source tree.

ThibaultDewailly commented:
Hello and welcome to this repository !

Please code the tests associated to the Pull request, in order to validate the behavior you are introducing

ThibaultDewailly reviewed (changes requested):
Blocker : Associated tests are mandatory

sinrazon commented:
As requested, I've reworked this PR to include related tests.

A few notes about these changes :

* @ThibaultDewailly suggested inverting the `$DEB_MAJ_VER` condition, to check for Debian 11 first, then sid; however, doing so does not fix the aforementioned error (string being tested against an integer), so this change was not made.
* There is no test utility to "mock" a Debian version, and the `$DEB_MAJ_VER` utility is outside of the individual test scope, which is not overridable. The accepted solution is to monkey-patch `/etc/debian_version` on the fly.
* A lot of tests were missing for both of these checks, so I've added them based on similar checks + new tests for Debian Sid.
* A couple of tweaks were needed to make the `apply()` function work for these scripts. I think the lack of sufficient testing broke them a while ago.

Please let me know if anything more is needed.
